code/modules/projectiles/gun.dm

```diff
@@ -1110,7 +1110,7 @@ and you're good to go.
 		var/obj/item/projectile/projectile_to_fire = load_into_chamber(user) //Load a bullet in or check for existing one.
 		if(!projectile_to_fire) //If there is nothing to fire, click.
 			click_empty(user)
-			break
+			return
```

Comment:
```suggestion
			flags_gun_features &= ~GUN_BURST_FIRING
			return
```
This broke guns _a lot_. It still looks wrong somehow, but at least this should stop guns freaking out dozen times per round.